src/lib.rs

```diff
@@ -9,33 +12,37 @@ pub use crate::chrono::avg_time_of_day;
 
 pub fn circadian_average<I, F>(range: F, data: I) -> (F, F)
 where
-    F: Float,
+    F: Float + Debug + Display,
     I: Iterator<Item = F>,
 {
     let mut len = 0;
     let mut x_pos_sum = F::zero();
     let mut y_pos_sum = F::zero();
-    
+
     // This unwrap is reasonable as this can not be done if F can't be represented as 2PI
     let tau = F::from(TAU).unwrap();
-    let dist_to_angle = tau / range;
+    let tau_over_range = tau / range;
 
     for x in data {
         len += 1;
         // Get X, Y position of each data point on a circle with a perimeter of range
-        let angle = x * dist_to_angle;
+        let angle = x * tau_over_range;
         let (s, c) = angle.sin_cos();
         x_pos_sum = x_pos_sum + c;
         y_pos_sum = y_pos_sum + s;
     }
 
     let avg_x_pos = x_pos_sum / F::from(len).unwrap();
     let avg_y_pos = y_pos_sum / F::from(len).unwrap();
-
     // Get the angle of the average position
-    let avg_angle = avg_y_pos.atan2(avg_x_pos);
+    let atan2 = avg_y_pos.atan2(avg_x_pos);
```

Comment:
The compiler will probably fold this anyhow, but I'd rephrase it as:
```rust
let mut avg_angle = avg_y_pos.atan2(avg_x_pos);
// atan2 range is -pi to +pi, thus we adjust negative values with tau 
if avg_angle < F::zero() {
   avg_angle += tau
}
```

Reply:
Good call.